File: cpp/34_find_first_and_last_position_of_element_in_sorted_array.c
class Solution {
    vector<int> nums;
    int target;
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        this->nums = nums;
        this->target = target;
        if(nums.empty())
            return {-1, -1};
        int index = this->binSearch(0, nums.size() - 1);
        if(index == -1)
            return {-1, -1};
        int i, j;
        i = j = index;
        while(-1 < i && nums[i] == target)
            --i;
        while(j < nums.size() && nums[j] == target)
            ++j;
        return {i + 1, j - 1};
    }
private:
    int binSearch(int left, int right){
        if(left > right)
            return -1;
        int mid = left + ((right - left) / 2);
        if(this->nums[mid] == this->target)
            return mid;
        if(this->nums[mid] > this->target)
            return binSearch(left, mid - 1);
        return binSearch(mid + 1, right);
    }
};

File: python/34_find_first_and_last_position_of_element_in_sorted_array.py
class Solution:
    def searchRange(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        def _binSearch(left, right):
            if left > right:
                return -1
            mid = left + ((right - left) // 2)
            if nums[mid] == target:
                return mid
            if nums[mid] > target:
                return _binSearch(left, mid - 1)
            return _binSearch(mid + 1, right)
        if nums == []:
            return [-1, -1]
        index = _binSearch(0, len(nums) - 1)
        if index == -1:
            return [-1, -1]
        i = j = index
        while -1 < i and nums[i] == target:
            i -= 1
        while j < len(nums) and nums[j] == target:
            j += 1
        return [i + 1, j - 1]